### PR TITLE
perf: search transition smoother on slower devices

### PR DESCRIPTION
**Pull request by hurali97**
### Details

When user navigates back from Search Sidebar there’s a noticeable delay and lag in closing the Search Sidebar. It mainly arises from `withNavigationFocus` HOC as it re-renders the `BaseOptionSelector` tree including `OptionsList` . Adding a `memo` to `OptionsList` doesn’t help as it’s parent gets re-rendered due to the `withNavigationFocus` HOC.
Also, we notice that due to `isFocused` state value in `ActionItemView` the `ReportActionsList` re-renders which also adds the overhead to the rendering cycle as it recreates the instances of `recordTimeToMeasureItemLayout` and `loadOlderChats` which are passed as props to `ReportActionsList` .

We can remove the `withNavigationFocus` HOC and add navigation listeners(focus, blur) in `BaseOptionSelector` and memoize `OptionsList` to not re-render when state is set in it’s parent. We can avoid recreating instances of `recordTimeToMeasureItemLayout` and `loadOlderChats` by wrapping them in `useCallbacks` and memoizing `ReportActionsList` .

<details>
<summary>Before on 6x CPU throttle:</summary>

https://github.com/Expensify/App/assets/47336142/84897de0-5f11-44ee-9b72-e67deb990bc0

_profile-ranked-before_

profiler-ranked-before

_profile-timeline-before_

profiler-ranked-after

</details>

<details>
<summary>After on 6x CPU throttle:</summary>

https://github.com/Expensify/App/assets/47336142/c9a27473-1234-4b05-a7dd-beb6d843ae2f

_profile-ranked-after_

profiler-ranked-after

_profile-timeline-after_

profiler-timeline-after

</details>

### Fixed Issues

$ https://github.com/Expensify/App/issues/33596 
PROPOSAL: https://expensify.slack.com/archives/C05LX9D6E07/p1699514971219659

### Tests

Manual Test:
- Click on Search Icon and Search Sidebar should appear quickly

- [X] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/47336142/23b1f8f3-b9f9-42bb-863c-f393904e2dbe

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/47336142/b7f56a46-2c71-43e9-ba9c-d88a8c37af24

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/47336142/14d85899-d1ef-4e20-b994-2512574d1377

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/47336142/93202430-b900-468e-9311-2f455ff44c1e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/47336142/a801348e-c508-4e0d-a439-eb4137129d6c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/47336142/8b936ab9-0e2a-4026-bb9c-f90f3716b4f2

</details>

### CODE REVIEW

File: src/components/OptionsList/BaseOptionsList.js [hunk 1]
```diff
@@ -35,276 +35,277 @@ const defaultProps = {
...
-            const sectionHeaderHeight = section.title && !hideSectionHeaders ? variables.optionsListSectionHeaderHeight : 0;
-            flatArray.push({length: sectionHeaderHeight, offset});
-            offset += sectionHeaderHeight;
-
-            // Add section items
-            for (let i = 0; i < section.data.length; i++) {
-                let fullOptionHeight = variables.optionRowHeight;
-                if (i > 0 && shouldHaveOptionSeparator) {
-                    fullOptionHeight += variables.borderTopWidth;
+const BaseOptionsList = forwardRef(
```
Comment (hurali97) [inline src/components/OptionsList/BaseOptionsList.js:38]: The main change here is refactoring the `forwardRef`. The second change is memoizing the `listContentContainerStyle `.

**Comment by situchan** [comment]
I can help if requires C+ review

**Comment by situchan** [comment]
I think @hurali97 is OOO this week. Maybe move this PR forward next week.

**Comment by hurali97** [comment]
@situchan conflicts resolved 👍

**Comment by Gonals** [comment]
> @situchan conflicts resolved 👍

I still see some

**Comment by hurali97** [comment]
> > @situchan conflicts resolved 👍
> 
> I still see some

Yeah it might be that when I fixed the previous, some latest changes were merged outside my working hours. I will resolve those as well.

**Comment by hurali97** [comment]
@Gonals conflicts resolved !!

**Comment by hurali97** [comment]
@situchan Are you able to review the PR? 🙂

**Comment by situchan** [comment]
yes reviewing today

**Comment by hurali97** [comment]
@situchan any updates on the review 🙂

**Comment by situchan** [comment]
reviewing now

File: src/pages/home/sidebar/SidebarLinks.js [hunk 2]
```diff
@@ -147,6 +147,10 @@ function SidebarLinks({onLinkClick, insets, optionListItems, isLoading, priority
     );
 
     const viewMode = priorityMode === CONST.PRIORITY_MODE.GSD ? CONST.OPTION_MODE.COMPACT : CONST.OPTION_MODE.DEFAULT;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const listStyle = useMemo(() => [isLoading ? styles.flexShrink1 : styles.flex1], [isLoading]);
```
[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/home/sidebar/SidebarLinks.js:151]: Oh my bad.. initially this was `style={[isLoading ? styles.flexShrink1 : styles.flex1]}` so at that time this memo was valid and now I checked with main there's only `style={styles.flex1}`. So I guess it happened in the merge conflict. I will revert this change, thanks for flagging this 👍

File: src/pages/SearchPage.js [hunk 3]
```diff
@@ -231,9 +253,6 @@ export default compose(
         reports: {
             key: ONYXKEYS.COLLECTION.REPORT,
         },
-        personalDetails: {
-            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-        },
```
[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/SearchPage.js:236]: Okay I will look into this, thanks 👍

[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/SearchPage.js:236]: @situchan I am trying to repro this but I can't. Can you tell me what shortcuts are you using or the repro steps if possible?


https://github.com/Expensify/App/assets/47336142/c0092f12-ac57-4b55-ab52-825718bfde6c

[the remark this answers is not in the document]
Comment (situchan) [reply src/pages/SearchPage.js:236]: Ah I tested on 2 devices. Let's say A has contact B!
- On A, search for B and keep search page open
- On B, update their display name (this should be done on another device)
- On A, B's profile is not updated on search page until close and re-open

I am reporting this because it doesn't happen on main

[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/SearchPage.js:236]: @situchan I am testing now on main as you have suggested using two different accounts on web and android emulator. However, on main the result is the same, the personal details doesn't update in real-time and you have to switch to the user and get the latest data for personal details.

Can you show me in a video, how it works for you on the main using two different accounts?

### Main:
https://github.com/Expensify/App/assets/47336142/d26739e5-ce91-4ef7-b879-f7cc48d613c0

[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/SearchPage.js:236]: I will test it one more time after clearing caches and verify Pusher is working.

[the remark this answers is not in the document]
Comment (hurali97) [reply src/pages/SearchPage.js:233]: Okay so I was able to repro this and it's now fixed 👍

**Comment by hurali97** [comment]
@situchan are you able to review the PR? 🙂 I have fixed the bug already 👍

**Comment by situchan** [comment]
yup, checking other similar instances

**Comment by hurali97** [comment]
@situchan Any leads on the testing ? 🙂

**Comment by situchan** [comment]
@hurali97 please pull main

**Comment by hurali97** [comment]
@situchan done 👍

**Comment by hurali97** [comment]
@situchan any updates on testing the PR? 🙂

**Comment by situchan** [comment]
Tests well so far. I will complete checklist today

**Comment by situchan** [comment]
I was waiting how search page optimization PR goes but it's actually reverted

**Comment by hurali97** [comment]
@situchan Oh .. so what's the proceedings now?

**Comment by situchan** [comment]
I think we can proceed with this one. But just to make sure it doesn't cause such regressions like #33282, #33291, #33322

**Comment by hurali97** [comment]
@situchan I get it. This doesn't seem relevant but you can run another regression to make sure we are safe.

I can sync with main before you proceed, sounds great?

**Comment by TMisiukiewicz** [comment]
This might fix #33326

**Comment by mountiny** [comment]
@situchan Could you prioritize this one even if there are conflicts now, maye try to resolve them locally yourself for testing purposes, but would be great to keep these going

**Comment by situchan** [comment]
The changes related to search page performance went back and forth as it affected HT accounts with lots of phone numbers.
I think we ready to test this again once conflicts are resolved.

**Comment by mountiny** [comment]
Yes happy to make another Build for this one so we can test with the internal users too

File: src/components/OptionsList/BaseOptionsList.js [hunk 4]
```diff
@@ -36,281 +36,283 @@ const defaultProps = {
...
+            highlightSelectedOptions,
+            onSelectRow,
+            boldStyle,
+            isDisabled,
+            isRowMultilineSupported,
+            isLoadingNewOptions,
+            nestedScrollEnabled,
+            bounces,
+            renderFooterContent,
+            safeAreaPaddingBottomStyle,
```
Comment (situchan) [inline src/components/OptionsList/BaseOptionsList.js:78]: What's `safeAreaPaddingBottomStyle` for? I don't see this in current `BaseOptionsList`

File: src/components/OptionsList/BaseOptionsList.js [hunk 5]
```diff
@@ -36,281 +36,283 @@ const defaultProps = {
...
+        },
+        innerRef,
+    ) => {
+        const styles = useThemeStyles();
+        const flattenedData = useRef();
+        const previousSections = usePrevious(sections);
+        const didLayout = useRef(false);
+
+        const listContainerStyles = useMemo(() => listContainerStylesProp || [styles.flex1], [listContainerStylesProp, styles.flex1]);
+        const contentContainerStyles = useMemo(() => [contentContainerStylesProp, safeAreaPaddingBottomStyle], [contentContainerStylesProp, safeAreaPaddingBottomStyle]);
```
Comment (situchan) [inline src/components/OptionsList/BaseOptionsList.js:88]:
```suggestion
        const contentContainerStyles = useMemo(() => [safeAreaPaddingBottomStyle, ...contentContainerStylesProp], [contentContainerStylesProp, safeAreaPaddingBottomStyle]);
```

Btw, any reason for moving `safeAreaPaddingBottomStyle` from parent component to here?

File: src/components/OptionsSelector/BaseOptionsSelector.js [hunk 6]
```diff
@@ -100,35 +99,46 @@ class BaseOptionsSelector extends Component {
...
                 this.unSubscribeFromKeyboardShortcut();
             }
-        }
+            this.isFocused = false;
+        });
+        this.scrollToIndex(this.props.selectedOptions.length ? 0 : this.state.focusedIndex, false);
 
-        // Screen coming back into focus, for example
-        // when doing Cmd+Shift+K, then Cmd+K, then Cmd+Shift+K.
-        // Only applies to platforms that support keyboard shortcuts
```
Comment (situchan) [inline src/components/OptionsSelector/BaseOptionsSelector.js:125]: Please restore this comment

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/108292595/7f656f8c-0675-4625-9b36-fbe717ce80c2



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/108292595/db5a482b-cff7-424f-b11f-697297e9c687



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/68d90aac-6aec-4407-b9d7-a1e258481f6d



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/108292595/435804c1-4ab5-4742-9513-07b54d7e43cf



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/108292595/ef5af3bc-0678-4fc7-97d8-3f3cd26221b4



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/108292595/29aa65c9-2108-45ed-955f-a9eab64965ca



</details>

File: src/components/OptionsList/BaseOptionsList.js (hunk 5 above)
Comment (hurali97) [reply src/components/OptionsList/BaseOptionsList.js:88]: The reason to move `safeAreaPaddingBottomStyle ` to child is to avoid the unnecessary re-render due to new array instance being created on each render. Below is how it was done previously in the parent component.

`contentContainerStyles={[safeAreaPaddingBottomStyle, ...this.props.contentContainerStyles]}`

Now we construct the `contentContainerStyles` in child component.

Comment (hurali97) [reply src/components/OptionsList/BaseOptionsList.js:88]: > Your current change will cause regression if contentContainerStylesProp includes paddingBottom.
i.e. TagPicker, ReportParticipantsPage

With your suggested change we can avoid this, right? I also see in my changes that I missed to spread the object, which you have highlighted.

File: src/components/OptionsSelector/BaseOptionsSelector.js (hunk 6 above)
Comment (hurali97) [reply src/components/OptionsSelector/BaseOptionsSelector.js:125]: Is it important? Since the code was refactored for which the comment was relevant.

File: src/components/OptionsList/BaseOptionsList.js (hunk 5 above)
Comment (situchan) [reply src/components/OptionsList/BaseOptionsList.js:88]: Just spreading is fine I think

File: src/components/OptionsSelector/BaseOptionsSelector.js (hunk 6 above)
Comment (situchan) [reply src/components/OptionsSelector/BaseOptionsSelector.js:125]: Just to avoid breaking this when someone updates code around here

**Comment by situchan** [comment]
Where's this coming?

Screenshot 2023-12-28 at 2 50 59 PM

**Comment by hurali97** [comment]
> Where's this coming?
> 
> Screenshot 2023-12-28 at 2 50 59 PM

Taking a look. It started happening after syncing the latest main. Wasn't happening previously, looks like some test in Reassure can't locate `useNavigation`.

**Comment by hurali97** [comment]
@situchan Can you maybe try re-triggering the Reassure test workflow? Looks like Github Action ran into some parsing issue. If we look at the output of Reassure there are no errors/ significant changes to the renders but an issue with Github Action itself.

https://github.com/Expensify/App/assets/47336142/1becc063-7e6c-448e-95f9-5b33924d6b7c

**Comment by situchan** [comment]
Please pull main again. Seems like not happening on other PRs

**Comment by situchan** [comment]
And revert https://github.com/Expensify/App/pull/31354/commits/94d9a50b8a00b43a28ed9eb90f33cf38d423b6dc if unnecessary

**Comment by hurali97** [comment]
> And revert https://github.com/Expensify/App/commit/94d9a50b8a00b43a28ed9eb90f33cf38d423b6dc if unnecessary

No, this is necessary. Reassure tests fails if we don't have this fix.

**Comment by situchan** [comment]
> > And revert [94d9a50](https://github.com/Expensify/App/commit/94d9a50b8a00b43a28ed9eb90f33cf38d423b6dc) if unnecessary
> 
> No, this is necessary. Reassure tests fails if we don't have this fix.

can you share context behind it? What change was failing that test?

**Comment by hurali97** [comment]
> > > And revert [94d9a50](https://github.com/Expensify/App/commit/94d9a50b8a00b43a28ed9eb90f33cf38d423b6dc) if unnecessary
> > 
> > 
> > No, this is necessary. Reassure tests fails if we don't have this fix.
> 
> can you share context behind it? What change was failing that test?

The fact that we have now Reassure test for `OptionsSelector-perf.test.js` and that PR have been merged recently and prior to this, the Reassure tests were not failing.

The issue arises due to `withNavigation` not being mocked and unavailability of `navigation` prop. Previously, `withNavigationFocus` was being mocked and since this PR uses `withNavigation` instead of `withNavigationFocus`, I have to mock the former and remove the latter.

File: src/components/OptionsList/BaseOptionsList.js [hunk 7]
```diff
@@ -36,282 +36,284 @@ const defaultProps = {
...
+        const buildFlatSectionArray = () => {
+            let offset = 0;
+
+            // Start with just an empty list header
+            const flatArray = [{length: 0, offset}];
+
+            // Build the flat array
+            for (let sectionIndex = 0; sectionIndex < sections.length; sectionIndex++) {
+                const section = sections[sectionIndex];
+                // Add the section header
```
Comment (mountiny) [inline src/components/OptionsList/BaseOptionsList.js:104]:
```suggestion

                // Add the section header
```

File: src/components/OptionsList/BaseOptionsList.js [hunk 8]
```diff
@@ -36,282 +36,284 @@ const defaultProps = {
...
+            const targetItem = flattenedData.current[flatDataArrayIndex];
+            return {
+                length: targetItem.length,
+                offset: targetItem.offset,
+                index: flatDataArrayIndex,
+            };
+        };
+
+        /**
+         * Returns the key used by the list
```
Comment (mountiny) [inline src/components/OptionsList/BaseOptionsList.js:174]:
```suggestion
         * Returns the key used by the list
         *
```

**Review by mountiny — DISMISSED** [review]
I think the failing tests are expected based on the explanation from @hurali97

**Review by situchan — APPROVED** [review]

**Comment by hurali97** [comment]
Coming from [here](https://callstack-hq.slack.com/archives/C05LX9D6E07/p1703844985106289?thread_ts=1703679928.194659&cid=C05LX9D6E07). Waiting on to fix Reassure Tests.

In the mean time, I will keep resolving conflicts so that we don't have huge conflicts at the end, when Reassure tests are fixed and we are following up again on this PR. Thanks.

**Comment by hurali97** [comment]
The reassure tests failure is expected as we have moved the following piece of code from being executed in `constructor` to execute once the transition animations are finished. Which results in a state update but providing a better user experience.

Screenshot 2024-01-08 at 4 12 27 PM

File: src/pages/home/report/ReportActionsList.js [hunk 9]
```diff
@@ -497,4 +497,4 @@ ReportActionsList.propTypes = propTypes;
 ReportActionsList.defaultProps = defaultProps;
 ReportActionsList.displayName = 'ReportActionsList';
 
-export default compose(withWindowDimensions, withPersonalDetails(), withCurrentUserPersonalDetails)(ReportActionsList);
+export default compose(withWindowDimensions, withPersonalDetails(), withCurrentUserPersonalDetails)(memo(ReportActionsList));
```
Comment (mountiny) [inline src/pages/home/report/ReportActionsList.js:500]: How is this change related to the rest of the PR?

**Review by mountiny — DISMISSED** [review]
Thanks! @Gonals Could you please review this one soon? thanks!

File: src/pages/home/report/ReportActionsList.js (hunk 9 above)
Comment (hurali97) [reply src/pages/home/report/ReportActionsList.js:516]: This goes hand in hand with the following:

> we notice that due to isFocused state value in ActionItemView the ReportActionsList re-renders which also adds the overhead to the rendering cycle as it recreates the instances of recordTimeToMeasureItemLayout and loadOlderChats which are passed as props to ReportActionsList

**Review by mountiny — APPROVED** [review]
Thanks!

**Review by Gonals — APPROVED** [review]

**Comment by Gonals** [comment]
As discussed in the PR, Performance tests failing here are not an emergency

**Comment by mountiny** [comment]
This was reverted as bunch of regressions have been found not related to the search page
